Physical_Engine/Query/SolidVolume.cs

```diff
@@ -136,7 +138,46 @@ public static double SolidVolume(this IOpening opening)
 
             return takeoff.SolidVolume();
         }
-        
+
+        /***************************************************/
+
+        [Description("Returns the solid volume of a PadFoundation based on the thickness and area.")]
+        [Input("padFoundation", "The PadFoundation to get the volume from.")]
+        [Output("volume", "The PadFoundation solid material volume.", typeof(Volume))]
+        public static double SolidVolume(this PadFoundation padFoundation)
+        {
+            if (padFoundation.IsNull())
+                return double.NaN;
```

Comment:
Seeing double.NaN being returned here I am wondering if not revisiting `EmbedVolume` method whether it shouldn't return `NaN` in any scenario?

Reply:
My preference would be to return zero and raise a warning - I was just replicating what had been done with the other methods.

Reply:
The problem with returning 0 @peterjamesnugent is that a user might use the method, get 0 because of an error, but that 0 can be legitimately used in downstream calculations. Even with the raising of a warning/error, users might not check it, or even be able to begin working out which pile from a hundred is causing the error, and the downstream calculation is calculating so no harm no foul right? That's how users think at least, so it's best to return something that cannot be an acceptable output from the method if something isn't working imo.

Reply:
Cool, happy with @FraserGreenroyd's response and will leave the code as is.